crates/macros/src/impl_.rs

```diff
@@ -88,172 +88,310 @@ pub struct AttrArgs {
+                                .args
+                                .typed
+                                .first()
+                                .map(|arg| arg.name == "self_")
+                                .unwrap_or_default()
+                            {
+                                // `self_: &[mut] ZendClassObject<Self>`
+                                // Need to remove arg from argument list
+                                func.args.typed.pop();
+                                MethodReceiver::ZendClassObject
+                            } else {
+                                // Static method
+                                MethodReceiver::Static
+                            },
+                        };
+                        let builder = func.function_builder(call_type)?;
+                        self.functions.push(FnBuilder {
+                            builder,
+                            vis: opts.vis,
+                            r#abstract: matches!(opts.ty, MethodTy::Abstract),
+                        });
                     }
-                    parsed_method.tokens
                 }
-                item => item.to_token_stream(),
-            })
-        })
-        .collect::<Result<Vec<_>>>()?;
+                _ => {}
+            }
+        }
+        Ok(())
+    }
 
-    let output = quote! {
-        impl #self_ty {
-            #(#tokens)*
+    /// Generates an `impl PhpClassImpl<Self> for PhpClassImplCollector<Self>`
+    /// block.
+    fn generate_php_class_impl(&self) -> Result<TokenStream> {
+        let path = &self.path;
+        let functions = &self.functions;
+        let constructor = match &self.constructor {
+            Some(func) => Some(func.constructor_meta(self.path)?),
+            None => None,
         }
-    };
+        .option_tokens();
+        let constants = self.constants.iter().map(|c| {
+            let name = &c.name;
+            let ident = c.ident;
+            let docs = &c.docs;
+            quote! {
+                (#name, &#path::#ident, &[#(#docs),*])
+            }
+        });
 
-    Ok(output)
-}
+        Ok(quote! {
+            impl ::ext_php_rs::internal::class::PhpClassImpl<#path>
+                for ::ext_php_rs::internal::class::PhpClassImplCollector<#path>
+            {
+                fn get_methods(self) -> ::std::vec::Vec<
+                    (::ext_php_rs::builders::FunctionBuilder<'static>, ::ext_php_rs::flags::MethodFlags)
+                > {
+                    vec![#(#functions),*]
+                }
 
-pub fn parse_attribute(attr: &Attribute) -> Result<Option<ParsedAttribute>> {
-    let name = attr.path.to_token_stream().to_string();
-    let meta = attr
-        .parse_meta()
-        .map_err(|_| anyhow!("Unable to parse attribute."))?;
+                fn get_method_props<'a>(self) -> ::std::collections::HashMap<&'static str, ::ext_php_rs::props::Property<'a, #path>> {
+                    todo!()
```

Comment:
It looks like this was never fixed, so actually getters / setters don't work at all right now